python_src/src/lamp_py/aws/ecs.py

```diff
@@ -41,3 +43,48 @@ def check_for_sigterm(
         time.sleep(5)
 
         sys.exit()
+
+
+def running_in_aws() -> bool:
+    """
+    return True if running on aws, else False
+    """
+    return bool(os.getenv("AWS_DEFAULT_REGION"))
+
+
+def check_for_parallel_tasks() -> None:
+    """
+    Check that that this task is not already running on ECS
+    """
+    if not running_in_aws():
+        return
+
+    process_logger = ProcessLogger("check_for_tasks")
+    process_logger.log_start()
+
+    client = boto3.client("ecs")
+    ecs_cluster = os.environ["ECS_CLUSTER"]
+    ecs_task_group = os.environ["ECS_TASK_GROUP"]
+
+    # get all of the tasks running on the cluster
+    task_arns = client.list_tasks(cluster=ecs_cluster)["taskArns"]
+
+    # if tasks are running on the cluster, get their descriptions and check to
+    # count matches the ecs task group.
+    match_count = 0
+    if task_arns:
+        running_tasks = client.describe_tasks(
+            cluster=ecs_cluster, tasks=task_arns
+        )["tasks"]
+
+        for task in running_tasks:
+            if ecs_task_group == task["group"]:
+                match_count += 1
+
+    # if the group matches, raise an exception that will terminate the process
+    if match_count > 1:
+        exception = SystemError(f"Multiple {ecs_cluster} ECS Tasks Running")
+        process_logger.log_failure(exception)
+        raise exception
+
+    process_logger.log_complete()
```

Comment:
might be worth putting this in a try except to check for key errors, log, and re-raise.

Reply:
Do you mean key errors for the `tasks` and `groups` indexing from `client.describe_tasks`?

Reply:
yeah. that would be the thing to catch i think? that way we get a failure log on this process rather than a failure on the main method.

not a huge deal, but feels a little cleaner and like a practice we should keep up imho.